### PR TITLE
Remove notes from upstream docs about storing secrets in plain text

### PR DESCRIPTION
**Pull request by mikhailshilkov**
>  All arguments including the following potentially sensitive values will be stored in the raw state as plain text

While this is true in terraform, that does not apply to Pulumi. My only change is removing lines from TF docs and then re-generating everything.

Fix https://github.com/pulumi/pulumi-gcp/issues/1195

### CODE REVIEW

File: patches/0006-docs-patching.patch [hunk 1]
```diff
@@ -664,38 +753,36 @@ index a2f6f69ba..2f7bc1ff3 100644
...
 +++ b/website/docs/r/compute_backend_service.html.markdown
-@@ -34,9 +34,8 @@ To get more information about BackendService, see:
+@@ -34,10 +34,6 @@ To get more information about BackendService, see:
  * How-to Guides
      * [Official Documentation](https://cloud.google.com/compute/docs/load-balancing/http/backend-service)
  
 -~> **Warning:** All arguments including the following potentially sensitive
 -values will be stored in the raw state as plain text: `iap.oauth2_client_secret`, `iap.oauth2_client_secret_sha256`.
 -[Read more about sensitive data in state](https://www.terraform.io/language/state/sensitive-data).
-+~> **Warning:** All arguments including `iap.oauth2_client_secret` and `iap.oauth2_client_secret_sha256` will be stored in the raw
```
Comment (VenelinMartinov) [inline patches/0006-docs-patching.patch:693]: Are these secrets? If so, isn't this still wrong?

Comment (mikhailshilkov) [reply patches/0006-docs-patching.patch:693]: > isn't this still wrong?

What do you mean?

Comment (mikhailshilkov) [reply patches/0006-docs-patching.patch:693]: Ah, I see, the note is still here... let me check why!

Comment (VenelinMartinov) [reply patches/0006-docs-patching.patch:693]: There's a few more notes like that, not just this one.
For example, `website/docs/r/iap_client.html.markdown`

Comment (mikhailshilkov) [reply patches/0006-docs-patching.patch:693]: This looks like a github rendering bug to me. There is no green text in the actual patch file.

Comment (VenelinMartinov) [reply patches/0006-docs-patching.patch:693]: Yeah, looks like the diff is buggy - when I viewed the whole file that text disappears.

Comment (mikhailshilkov) [reply patches/0006-docs-patching.patch:693]: You can see the note is gone e.g. in sdk/dotnet/Iap/Client.cs below

**Review by iwahbe — CHANGES_REQUESTED** [review]
Can we use a rule to remove this instead of a patch?

This greatly expands the size of the docs patch. I'm also concerned that this will not be evenly applied to all resources (especially new resources).

We can do that by adding another docs edit rule:

https://github.com/pulumi/pulumi-gcp/blob/9edf26fac0772e0b87ad96f5342f9e171922cae6/provider/doc_edits.go#L9-L12

**Comment by iwahbe** [comment]
The pattern to remove looks like:

```
~> **Warning:** All arguments including the following potentially sensitive
values will be stored in the raw state as plain text: .*\\.
[Read more about sensitive data in state](https://www.terraform.io/language/state/sensitive-data).

```

**Comment by mikhailshilkov** [comment]
@iwahbe I thought about that but the pattern won't be reliable either. As an example, the patch is a mix of snippets with "plain-text" and "plain text", so your pattern above won't catch the former options. Also, there was one edit that I made to exlcude the secrety text but keep the import text in the same paragraph.

I'll give it a try if I can find 2-3 patterns that catch (almost?) all instances.

**Comment by iwahbe** [comment]
> @iwahbe I thought about that but a single pattern won't be reliable either. As an example, the patch is a mix of snippets with "plain-text" and "plain text", so your pattern above won't catch the former options. Also, there was one edit that I made to exlcude the secrety text but keep the import text in the same paragraph.
> 
> I'll give it a try if I can find 2-3 patterns that catch (almost?) all instances.

It can be as complex as an arbitrary function, but I think a regex will work pretty well here. One or two special cases handled via patch are ok. I don't want to take a patch that will need to be updated regularly when it seems like it could be avoided. If you can't find a pattern that will fix (most of) this, then we'll take the patch.

`"plain[ -]text"` will work on the example that you mentioned.

**Pull request closed without merging**